File: 121. Best Time to Buy and Sell Stock.cpp
#include<iostream>
#include<vector>
using namespace std;
class Solution {
public:
	int maxProfit(vector<int>& prices) {
		if (prices.empty())
			return 0;
		int maxProfit = 0;
		int min = prices[0];
		for (int i = 1; i < prices.size(); i++){
			if (prices[i] - min > maxProfit)
				maxProfit = prices[i] - min;
			if (prices[i] < min)
				min = prices[i];
		}
		return maxProfit;
	}
};

File: java/Array/121. Best Time to Buy and Sell Stock.java
public class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = 0, cur;
        if(prices.length == 0) return 0;
        else cur = prices[0];

        for(int i: prices) {
            if(i - cur > maxProfit) {
                maxProfit = i - cur;
                continue;
            }
            if(i < cur) cur = i;
        }

        return maxProfit;
    }
}
